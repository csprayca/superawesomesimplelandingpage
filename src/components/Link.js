'use strict';

const React = require('react');

const Link = ({ title, fromURL }) => {
  // if (title && title.length === 0) {
  //   return null;
  // }
  // if (fromURL && fromURL.length === 0) {
  //   return null;
  // }

  return (
    <React.Fragment>
      <a href={fromURL} role="button" title={title}>
        {title}
      </a>
      <style jsx>{`
        a {
          text-decoration: none;
          width: 100%;
          max-width: 100%;
          word-wrap: break-word;
          padding: 1em;
          border-width: 2px;
          border-style: solid;
          border-radius: 1em;
          text-align: center;
        }
      `}</style>
    </React.Fragment>
  );
};

exports = module.exports = Link;
