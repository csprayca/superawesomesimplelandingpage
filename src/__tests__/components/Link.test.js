/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../../components/Link';

const expected = {
  title: 'Youtube',
  toURL: 'https://www.youtube.com/channel/UCB5mFx3KjuOuMZoXDLSUvSg',
  fromURL: '/youtube'
};

describe('Link Component', () => {
  describe('given valid input', () => {
    test('works as expected', () => {
      const component = renderer.create(
        <Link title={expected.title} fromURL={expected.fromURL} />
      );

      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('given missing input', () => {
    test('it renders with no title when a title is missing', () => {
      const component = renderer.create(<Link />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('it renders with no target link when a fromURL is missing', () => {
      const component = renderer.create(<Link title={expected.title} />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
