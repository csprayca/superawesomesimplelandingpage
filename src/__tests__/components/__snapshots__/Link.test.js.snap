// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component given missing input it renders with no target link when a fromURL is missing 1`] = `
<a
  className="jsx-2004290176"
  role="button"
  title="Youtube"
>
  Youtube
</a>
`;

exports[`Link Component given missing input it renders with no title when a title is missing 1`] = `
<a
  className="jsx-2004290176"
  role="button"
/>
`;

exports[`Link Component given valid input works as expected 1`] = `
<a
  className="jsx-2004290176"
  href="/youtube"
  role="button"
  title="Youtube"
>
  Youtube
</a>
`;
